### ISSUE TITLE
Web clipper feature: option to automatically clip every page on load

### ISSUE THREAD

**Issue by rawwerks** [obsidianmd/obsidian-clipper#275]
TL;DR - I want to browse and trust that the clipper (& interpreter) are taking notes for me.

### Use case or problem

Obsidian Web clipper is great. So great that I'd like the option to have it "clip" every website I visit automatically. 

### Proposed solution

In the Extension Settings, an option that would be default Off for privacy, but the user could select to have every page automatically clipped on page load. 

The one potential complication (which I've experienced with Hunchly) is for pages with dynamic content that doesn't change the URL. So for example, if I'm looking at the X/Twitter homepage feed, it can be tricky to tell the plugin how to add/overwrite the clip when the content changes but the URL doesn't. I think the feature would be useful even if this were too challenging to solve.

### Current workaround (optional)

I am currently evaluating Hunchly as a workaround.

I am also looking into ways I can configure Chrome to "script" Obsidian Web Clipper.

### Related feature requests (optional)

Now that the Web Clipper has Interpreter, this would be a very powerful "automatic note-taking" workflow. Basically, I just browse and trust that the clipper & interpreter are taking notes for me.

**Comment by kepano** [comment]
I don't think we should add this option for many reasons. It would require some additional permissions that I don't think are necessary. But also I don't think any changes to Web Clipper are necessary in order for you to achieve this. What you need to do is automate triggering the "Quick clip" action.

Keep in mind that doing this will be a slow and expensive way to browse. It's probably better to build a more dedicated extension for this purpose.

**Comment by rawwerks** [comment]
ok, thanks for reviewing. 

_> "What you need to do is automate triggering the "Quick clip" action."_

any tips on how to accomplish this in chrome?

**Comment by AarooneyFarr** [comment]
I am also interested in this functionality. It looks like the plugin is using the browser command and messages API to send the "Quick clip" action to the correct place in the plugin. 

I am very new to plugins, so it is possible I have misunderstood this, but the onMessage listener only listens for messages coming from the same extension. One potential solution would be to add a few lines of code with the onMessageExternal, which would do the same thing as the onMessage, but open up the message API so actions could be triggered by another extension.

Currently the quick clip is triggered by a keyboard shortcut, making use of the command API that allows plugins to bind a function to a keyboard shortcut. I have spent most of the day looking for a way to automate this, but I could not find anything. Injected user scripts are sandboxed, so no code based injection will properly trigger. There are also basically no tools for triggering a keyboard shortcut automatically.

@kepano, it is possible I have misunderstood this, as I am a brand new developer. I am willing to try to PR changes, but this is my first attempt at contributing to open source, so I may need a little guidance.

**Comment by AarooneyFarr** [comment]
@rawwerks I figured out a way to trigger the plugin using puppeteer. Was able to get the functionality you are looking for to work.

**Comment by rawwerks** [comment]
@AarooneyFarr - sweet! Would you be willing to share the details? 

I'm trying to get the "browser-use" community to help me give AI access to plugins, I think it's also possible w playwright.

**Comment by AarooneyFarr** [comment]
@rawwerks I am sorry it took so long to respond here! I did not see the mention. Here is a write-up of my process. I was using the clipper for bulk web scraping, so some aspects of the functionality may need to be modified to better suite your use case, but hopefully it is a little bit helpful if you are still working on this. https://github.com/AarooneyFarr/web-scraper-description/blob/main/Recipe%20scraping.md
